modules/doc/content/ncrc/applications/ncrc_develop.md.template

```diff
@@ -0,0 +1,147 @@
+[NCRC/{{ApplicationName}}](ncrc/applications/ncrc_level4_{{ApplicationLower}}.md).
+
+## Building {{ApplicationName}} id=build
+
+Building {{ApplicationName}} begins with applying upstream changes, updating your submodules and/or
+Conda packages, and finally running `make`. Before we begin, +remember these two important
+warnings+:
+
+!alert warning title=You updated Conda?
+`Then you should also perform a submodule update`.
+
+!alert warning title=You performed a submodule update?
+`Then you should also perform a Conda update or rebuild PETSc/libMesh`.
+
+The symmetry importance comes from our Conda packages and support libraries being version controlled
+by and within the MOOSE repository. Updates to MOOSE (the `{{binary}}/moose` submodule) requires you
+to stay on top of any Conda updates (if Personal Machine), or rebuilding PETSc and libMesh (if using
+[!ac](HPC) resources).
+
+### Applying Upstream Changes
+
+If this is your first time following these instructions there should be no updates. However, we will
+perform an update to familiarize yourself with the entire process.
+
+{{ApplicationName}} developers will expect users to have performed the following sections, whenever
+they says "I have updated {{ApplicationName}}".
+
+```bash
+cd ~/projects/{{binary}}
+git fetch upstream
+```
+
+`git fetch upstream` updates your local references with that of the remote. It does not actually
+modify any source files. It will return nothing if there is nothing new to update.
+
+You need to know your upstream head branch. This will be the branch you will want to continuously
+rebase against. That is, keeping your local {{binary}} repository up to date. The following command
+will reveal the upstream head branch name:
+
+```bash
+git remote show upstream | grep "HEAD branch" | sed 's/.*: //'
+```
+
+Next, we will perform a `rebase` against the provided upstream branch name:
+
+```bash
+git rebase upstream/branch   # replace 'branch' with the discovered branch name
+```
+
+With {{binary}} updated, we can update its submodules.
+
+### Updating Submodules and Conda Packages
+
+Most MOOSE-based applications require additional support libraries available in the form of
+git submodules. To update submodules, perform the following:
+
+```bash
+cd ~/projects/{{binary}}
+git submodule update --init
+```
+
+Remember the cyclical warnings earlier? By performing a submodule update, you should now perform
+either a Conda update, or rebuilding PETSc/libMesh:
+
+- If +Personal Machine (Conda)+:
+
+  ```bash
+  mamba activate moose
+  mamba update --all
+  ```
+
+- If +[!ac](INL) [!ac](HPC) Sawtooth or Lemhi+:
+
+  It will be necessary to build PETSc and libMesh using the supplied build scripts in
+  `moose/scripts`. The time it takes to build both PETSc and libMesh can be hours:
+
+  ```bash
+  module load use.moose moose-dev    # a reminder to have these loaded
+  export MOOSE_JOBS=6 METHODS=opt    # a reminder to set these
```

Comment:
I'd probably delete these comments too. I think any time it's a console command rather than lines of a script, you should avoid comments.

Reply:
The reason these are here, is that users all the time forget to have these loaded. Logan and I then receive an email forwarded from the NCRC guys, from the customer... and this was the issue.

Reply:
Removed these comments. Its harmless to repeat the commands if they previously ran them.